### DISCUSSION TITLE
if it is possible to change MotifMatrix for correlateMatrices

### DISCUSSION THREAD

**Discussion by wangmhan** [GreenleafLab/ArchR#725] [Feature Requests]
Hi,

This is a useful package, thank you.

I met problems when consider to identify positive TF regulators. 
1) One problem is the inconsistent of gene name between TFs in MotifMatrix and gene name in GeneScoreMatrix and GeneIntegrationMatrix. As I am working with chicken, there is no specific motif database for it. So I used addMotifAnnotations with JASPAR2020, tax_group = "vertebrates". As they are conserved motifs, so some gene name are not same with chicken genome, i.e. with "()" TFAP2B(var.2), or  share same motifs Smad2::Smad3. All these motifs with "()" and "::" are removed for following analysis, but some are important and I would like to have them. If I can change the motif name: remove "()", and put the ones with "::" into two rows, then I can get around 100 more TFs. Is there a easy way to modify MotifMatrix and put it back into ArchR project? Or any other solutions?

2) as you mentioned in the paper, the aggregates are adopted from cicero. If I understand correct, in cicero, by default are using the KNN from UMAP projections. But here, in examples, are using LSI dimensions? Is it because you have did some analysis and found using LSI is better? Also, in Signac, they removed the first dimension of LSI as it is highly correlated with depth. But in ArchR seems still include first dimension for following analysis, why?

Thank you! Looking forward to the reply!

**Comment by rcorces** [comment]
1. I dont have any insight here. This is a use case that we dont run into.
2. ArchR automatically tests each dimension for correlation to sequencing depth and excludes dimensions that are highly correlated based on user input to the `corCutoff` parameter. This is a much more appropriate way of handling this problem. Also the dimensionality reduction implementations are not the same (described in the paper) and so the first dimension in ArchR is often not highly correlated to depth. In general projecting into the LSI subspace is going to be much more robust than projecting into the UMAP subspace.

**Comment by wangmhan** [comment]
Thank you for the quick reply!

It is a case issue, but others who work with non model organisms can also
meet the same problem. It is caused by the inconsistent
between gene name of MotifMatrix and GeneScoreMatrix. If the MotifMatrix
can be custom modified and put back, then it can be solved.
The second question is clear now, thanks.

**Comment by rcorces** [comment]
What about an implementation where the end-user has to create their own motif catalog with their own names?

**Comment by wangmhan** [comment]
Hi,

Yes, sounds good!
It would be nicer if could have some instructions on how to create the
motif catalog. As today I tried to add my modified motif annotation via
function addMotifAnnotations(), it goes well. But when I checked the
MotifMatrix via getMatrixFromProject(), I found the rowname is motifID
instead of motif related TF gene name...
